src/main/scala/org/broadinstitute/hail/sparkextras/OrderedRDD.scala

```diff
@@ -422,6 +422,61 @@ class OrderedRDD[PK, K, V] private(rdd: RDD[(K, V)], val orderedPartitioner: Ord
       new OrderedRDD[PK, K, V](new BlockedRDD(rdd, newPartEnd), partitioner)
     }
   }
+
+  def filterIntervals(intervals: IntervalTree[PK]): OrderedRDD[PK, K, V] = {
+
+    import kOk.pkOrd
+    import kOk._
+    import Ordering.Implicits._
+
+    if (partitions.length <= 1) // FIXME: hacky
+      return this
+
+    if (intervals.isEmpty)
+      return OrderedRDD.empty[PK, K, V](rdd.sparkContext)
+
+
+    val intervalArray = intervals.toArray
+
+    val partitionIndexBuilder = mutable.ArrayBuilder.make[Int]
+
+    val rangeBounds = orderedPartitioner.rangeBounds
+
+    val nPartitions = partitions.length
+    var i = 0
+    while (i < nPartitions) {
+      if (i == 0) {
+        if (intervalArray.exists(_.start <= rangeBounds(0)))
+          partitionIndexBuilder += 0
+      } else if (i == nPartitions - 1) {
+        if (intervalArray.reverse.exists(_.end > rangeBounds.last))
+          partitionIndexBuilder += i
+      } else {
+        val lastMax = rangeBounds(i - 1)
+        val thisMax = rangeBounds(i)
+        // FIXME: loads a partition if the lastMax == interval.start.  Can therefore load unnecessary partitions
+        if (intervals.overlaps(Interval(lastMax, thisMax)) || intervals.contains(thisMax))
```

Comment:
range bounds exist for the first `n - 1` partitions.  The rule is that elements in your partition fall in: `rangeBounds(last) < element <= rangeBounds(this)`